pkg/distsql/select_result.go

```diff
@@ -381,6 +415,16 @@ func (r *selectResult) fetchResp(ctx context.Context) error {
 		if err != nil {
 			return errors.Trace(err)
 		}
+
+		if len(r.selectResp.IntermediateOutputs) != len(intermediateOutputTypes) {
+			return errors.Errorf(
```

Comment:
Does it indicates inconsistency? If so the `inconsistencyReporter` could be used to log necessary information.

Reply:
This situation should never happen, but it has no relationship with the data consistency. If it happens, it is more like the TiDB or TiKV code has a bug that makes the wrong assumption of the data channels.